graveler/committed/range_manager.go

```diff
@@ -77,6 +77,10 @@ type RangeWriter interface {
 	// WriteRecord appends the given record to the Range.
 	WriteRecord(record Record) error
 
+	// AddMetadata associates metadata value (which will be stringified when the writer is
+	// Closed) with key.
+	AddMetadata(key string, value interface{})
```

Comment:
if they end up getting stringified I suggest changing the signature to accept either a byte slice, or better yet, a string. for many types, the string representation will be either lossy or simply unparsable.

Reply:
Done, although it makes writing numbers a bit nasty.